### PR TITLE
Make liveness tracking of `LazyTensorOperations` a thread local state.

### PR DESCRIPTION
**Pull request by bgogul**

### CODE REVIEW

File: Sources/TensorFlow/Core/LazyTensorContext.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/// A class to keep track of runtime information about `LazyTensorOperation`
+/// instances created by the program. This will be manaaged as a thread local
```
Comment (pschuh) [inline Sources/TensorFlow/Core/LazyTensorContext.swift]: manaaged ?

**Comment by rxwei** [comment]
You could put all thread-local state into `Context` as private/internal properties, and deprecate `_ThreadLocalState`.

File: Sources/TensorFlow/Core/LazyTensorContext.swift (hunk 1 above)
Comment (bgogul) [reply Sources/TensorFlow/Core/LazyTensorContext.swift]: ah. fixed now. :)

**Comment by bgogul** [comment]
> You could put all thread-local stuff into `Context` as private/internal properties, and deprecate `_ThreadLocalState`.

Good idea. Let me do that in a follow up PR.

**Pull request merged**